hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -82,20 +82,17 @@ public final class HoodieFileGroupReader<T> implements Closeable {
   private ClosableIterator<T> baseFileIterator;
   private final Option<UnaryOperator<T>> outputConverter;
   private final HoodieReadStats readStats;
+  // Allows to consider inflight instants while merging log records using HoodieMergedLogRecordReader
+  // The inflight instants need to be considered while updating RLI records. RLI needs to fetch the revived
+  // and deleted keys from the log files written as part of active data commit. During the RLI update,
+  // the allowInflightInstants flag would need to be set to true. This would ensure the HoodieMergedLogRecordReader
```

Comment:
So this flag `allowInflightInstants` can be eliminated after the DAG write of DT and MDT? cc @nsivabalan

Reply:
yes, you are right.

Reply:
I'm OK with this flag temporarily, but as @danny0405 mentioned we should really get rid of such unorthodox read logic that is only due to bandaid, after rewriting of MDT write path.